server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/StarTreesBuilder.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.builder;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.codecs.DocValuesConsumer;
+import org.apache.lucene.codecs.DocValuesProducer;
+import org.apache.lucene.index.SegmentWriteState;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeField;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeFieldConfiguration;
+import org.opensearch.index.mapper.MapperService;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+public class StarTreesBuilder implements Closeable {
+
+    private static final Logger logger = LogManager.getLogger(StarTreesBuilder.class);
+
+    private final List<StarTreeField> starTreeFields;
+    private final StarTreeFieldConfiguration.StarTreeBuildMode buildMode;
+    private final DocValuesConsumer docValuesConsumer;
+    private final SegmentWriteState state;
+    private final DocValuesProducer docValuesProducer;
+    private final MapperService mapperService;
+
+    public StarTreesBuilder(
+        List<StarTreeField> starTreeFields,
+        StarTreeFieldConfiguration.StarTreeBuildMode buildMode,
+        DocValuesProducer docValuesProducer,
+        DocValuesConsumer docValuesConsumer,
+        SegmentWriteState segmentWriteState,
+        MapperService mapperService
+    ) {
+        this.starTreeFields = starTreeFields;
+        if (starTreeFields == null || starTreeFields.isEmpty()) {
+            throw new IllegalArgumentException("Must provide star-tree builder configs");
```

Comment:
nit : Can we change the loggers - configs are not present anymore

Also what's the plan for file formats, Do we need to create file formats here ? since we are calling build for each field and we'll be using the same file across fields.

In that case, do we need to move this to the lucene/index package ?

Reply:
As discussed offline, we might have to initialize our file formats over here and pass it on to BaseSingleTreeBuilder. Since we are in experimental, it would be okay to break the ctors for this change.